File: src/examples/java/httpperf/vertx-server.js
load('vertx.js')

vertx.createHttpServer().requestHandler(function(req) {
  //req.response.end();
  //req.response.sendFile("httpperf/foo.html");

  vertx.fileSystem.readFile("httpperf/foo.html", function(err, file) {
    req.response.headers["Content-Length"] = file.length()
    req.response.headers["Content-Type"] = "text/html"
    req.response.end(file)
  });
}).listen(8080, 'localhost');

File: vertx-examples/src/main/java/httpperf/vertx_server.rb
require "vertx"
include Vertx

HttpServer.new.request_handler do |req|
  # req.response.end
  # req.response.send_file("httpperf/foo.html")
  FileSystem.read_file_as_buffer("httpperf/foo.html") { |err, file|
    req.response.headers["Content-Length"] = file.length
    req.response.headers["Content-Type"] = "text/html"
    req.response.write_buffer(file)
    req.response.end
  }
end.listen(8080)
